### PR TITLE
Add overloaded method for adding empty parameters

### PR DESCRIPTION
**Pull request by domire8**
## Description

- #24 

As requested, I'm adding another method for adding parameters, specifically adding an empty parameter. There's not much more to that feature except the comment below. Nothing to add in python because there you have to use the Parameter constructor anyway.

## Review guidelines

Estimated Time of Review: 5 minutes

### CODE REVIEW

File: source/modulo_components/test/cpp/test_component_interface_parameters.cpp [hunk 1]
```diff
@@ -67,6 +69,22 @@ TYPED_TEST(ComponentInterfaceParameterTest, AddNameValueParameter) {
...
+  EXPECT_THROW(auto discard = this->component_->get_parameter("test"), exceptions::ComponentParameterException);
+  EXPECT_THROW(this->component_->get_ros_parameter("test"), rclcpp::exceptions::ParameterNotDeclaredException);
+  this->component_->template add_parameter<int>("test", "Test parameter");
+
+  // Adding the parameter should declare and set the value and call the validation function
+  EXPECT_TRUE(this->component_->validate_parameter_was_called);
+  EXPECT_NO_THROW(auto discard = this->component_->get_parameter("test"));
+  EXPECT_NO_THROW(this->component_->get_ros_parameter("test"));
+  EXPECT_STREQ(this->component_->get_ros_parameter("test").value_to_string().c_str(), "not set");
+  EXPECT_THROW(this->component_->template get_parameter_value<int>("test"),
```
Comment (domire8) [inline source/modulo_components/test/cpp/test_component_interface_parameters.cpp:82]: Should this maybe throw a ComponentParameterError

**Comment by domire8** [comment]
Humble tests don't pass anymore because `make_shared_parameter<T>(name)` is a feature on develop.

**Review by eeberhard — APPROVED** [review]

File: source/modulo_components/test/cpp/test_component_interface_parameters.cpp (hunk 1 above)
Comment (domire8) [reply source/modulo_components/test/cpp/test_component_interface_parameters.cpp:82]: Wait now I'm a bit confused. You want to add an `EmptyParameterException` to state representation?

I was suggesting throwing a `ComponentParameterException` (already exists) such that any component exception can be caught with the base exception class `ComponentException` but I agree with you that we should change anything here.

Comment (eeberhard) [reply source/modulo_components/test/cpp/test_component_interface_parameters.cpp:82]: I was mainly agreeing that accessing an empty parameter is not the component's fault, and so should not be a `ComponentException` or `ComponentParameterException`. Rather it's a state representation exception, so catching just to rethrow at component scope doesn't make sense imo. 

But I just added the related thought that, if you wanted to be more specific about the emptiness, another exception class in state representation could potentially be nice. It's not needed and changes nothing except making empty parameters maybe a little friendlier to catch

Comment (domire8) [reply source/modulo_components/test/cpp/test_component_interface_parameters.cpp:82]: I see yes, thanks

**Pull request merged**